clickhouse/columns/lowcardinality.cpp

```diff
@@ -174,6 +174,12 @@ ColumnLowCardinality::ColumnLowCardinality(std::shared_ptr<ColumnNullable> dicti
 ColumnLowCardinality::~ColumnLowCardinality()
 {}
 
+void ColumnLowCardinality::Reserve(size_t new_cap)
+{
+    dictionary_column_->Reserve(new_cap);
```

Comment:
Here we assume that **ALL** of the new items are unique, wich is quite an uncommon and quite sub-optimal case for `LowCardinality`. 

Maybe estimate dict size as `ln(new_cap) * sqrt(new_cap) + new_cap / 5`, this way we'll get:

| new_cap    | dict's new_cap      | estimated % of unique values |
|------------|---------------------|---------|
| 10 	| 10 	| 100.0  |
| 20 	| 18 	| 90.0  |
| 40 	| 32 	| 80.0  |
| 60 	| 44 	| 73.33  |
| 80 	| 56 	| 70.0  |
| 100 	| 67 	| 67.0  |
| 200 	| 115 	| 57.5  |
| 400 	| 200 	| 50.0  |
| 600 	| 277 	| 46.17  |
| 800 	| 350 	| 43.75  |
| 1000 	| 419 	| 41.9  |
| 2000 	| 740 	| 37.0  |
| 4000 	| 1325 	| 33.12  |
| 6000 	| 1874 	| 31.23  |
| 8000 	| 2404 	| 30.05  |
| 10000 	| 2922 	| 29.22  |
| 20000 	| 5401 	| 27.01  |
| 40000 	| 10120 	| 25.3  |
| 60000 	| 14695 	| 24.49  |
| 80000 	| 19194 	| 23.99  |
| 100000 	| 23641 	| 23.64  |
| 1000000 	| 213816 	| 21.38  |
| 10000000 	| 2050970 	| 20.51  |

Reply:
Actually, I do not understand the form, so could you  implement ColumnLowCardinality::Reserve ? 🌹

Reply:
Sure, will do